go-proio/reader.go

```diff
@@ -120,33 +127,41 @@ var (
 // the beginning of an event, the stream will be resynchronized to the next
 // event, and ErrResync will be returned along with the event.
 func (rdr *Reader) Get() (*Event, error) {
+	rdr.getMutex.Lock()
+
```

Comment:
you might as well use `defer rdr.getMutex.Unlock()` here instead of sprinkling it everywhere :)

Reply:
I actually was just about to say that too, thanks :)